tests/meson.build

```diff
@@ -0,0 +1,108 @@
+project('libnpy-tests', 'cpp',
+        default_options : ['cpp_std=c++14', 'warning_level=3'])
+
+pymod = import('python')
+python = pymod.find_installation('python3', required : true)
+
+
+catch2_dep = dependency('catch2', required : false)
+if not catch2_dep.found()
+  catch2_proj = subproject('catch2')
+  catch2_dep = catch2_proj.get_variable('catch2_dep')
+endif
```

Comment:
Depending on the version of Meson you're willing to require as a minimum:
- if you try building with Meson 0.55.0, `dependency('catch2')` automatically uses the subproject, with *no* additional work required
- if you try building with Meson older than 0.55.0, use `dependency('catch2', fallback: ['catch2', 'catch2_dep'])` to tell Meson how to find it.


The second method ("fallback" kwarg) is needed because the
```ini
[provide]
catch2 = catch2_dep
```
in the wrap file is new since Meson 0.55.0.

Either way, you do not need to check for `catch2_dep.found()` and then manually initialize the subproject.

Reply:
Thanks for the advice! Changed with 4df0150